### PR TITLE
fix issue with conditional and symbolic parameter in conversion

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Fix https://github.com/CQCL/pytket-qiskit/issues/43 and https://github.com/CQCL/pytket-qiskit/issues/224

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: tests/qiskit_convert_test.py [hunk 1]
```diff
@@ -413,7 +440,7 @@ def test_condition_errors() -> None:
         c = Circuit(2, 2)
         c.X(0, condition_bits=[1, 0], condition_value=1)
         tk_to_qiskit(c)
-    assert "OpenQASM conditions must be an entire register in order" in str(
+    assert "OpenQASM conditions must be an entire register in\n order" in str(
```
Comment (cqc-alec) [inline tests/qiskit_convert_test.py:443]: Don't much like having the newline in the error message.

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 2]
```diff
@@ -653,22 +666,31 @@ def append_tk_command_to_qiskit(
...
-                )
         instruction = append_tk_command_to_qiskit(
             op.op, args[width:], qcirc, qregmap, cregmap, symb_map, range_preds  # type: ignore
         )
+        if len(cregmap[regname]) == width:
+            for i, a in enumerate(args[:width]):
+                if a.index != [i]:
+                    raise NotImplementedError(
+                        """OpenQASM conditions must be an entire register in
+ order or only one bit of one register"""
```
Comment (cqc-alec) [inline pytket/extensions/qiskit/qiskit_convert.py:682]: Can we make this a string with no newlines?

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 3]
```diff
@@ -653,22 +666,31 @@ def append_tk_command_to_qiskit(
...
+                    )
+
+            instruction.c_if(cregmap[regname], value)
+        elif width == 1:
+            instruction.c_if(cregmap[regname][args[0].index[0]], value)
+        else:
+            raise NotImplementedError(
+                """OpenQASM conditions must be an entire register in
+order or only one bit of one register"""
```
Comment (cqc-alec) [inline pytket/extensions/qiskit/qiskit_convert.py:691]: Can we make this a string with no newlines?

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 4]
```diff
@@ -719,7 +741,10 @@ def append_tk_command_to_qiskit(
         ) from error
     params = _get_params(op, symb_map)
     g = gatetype(*params)
-    qcirc.global_phase += phase * sympy.pi
+    if type(phase) == float:
+        qcirc.global_phase += float(phase * sympy.pi)
```
Comment (cqc-alec) [inline pytket/extensions/qiskit/qiskit_convert.py:745]: May as well use `numpy.pi` or `math.pi` here?

**Review by cqc-alec — COMMENTED** [review]
Just small comments.

File: pytket/extensions/qiskit/qiskit_convert.py (hunk 3 above)
Comment (cqc-melf) [reply pytket/extensions/qiskit/qiskit_convert.py]: done in @cqc-melf
[remove linebreak from error message](https://github.com/CQCL/pytket-qiskit/pull/245/commits/f2a4770491ea359b7c93e021be5d46220611e92a)
[f2a4770](https://github.com/CQCL/pytket-qiskit/pull/245/commits/f2a4770491ea359b7c93e021be5d46220611e92a)

File: pytket/extensions/qiskit/qiskit_convert.py (hunk 2 above)
Comment (cqc-melf) [reply pytket/extensions/qiskit/qiskit_convert.py]: done in @cqc-melf
[remove linebreak from error message](https://github.com/CQCL/pytket-qiskit/pull/245/commits/f2a4770491ea359b7c93e021be5d46220611e92a)
[f2a4770](https://github.com/CQCL/pytket-qiskit/pull/245/commits/f2a4770491ea359b7c93e021be5d46220611e92a)

File: tests/qiskit_convert_test.py (hunk 1 above)
Comment (cqc-melf) [reply tests/qiskit_convert_test.py]: done in @cqc-melf
[remove linebreak from error message](https://github.com/CQCL/pytket-qiskit/pull/245/commits/f2a4770491ea359b7c93e021be5d46220611e92a)
[f2a4770](https://github.com/CQCL/pytket-qiskit/pull/245/commits/f2a4770491ea359b7c93e021be5d46220611e92a)

File: pytket/extensions/qiskit/qiskit_convert.py (hunk 4 above)
Comment (cqc-melf) [reply pytket/extensions/qiskit/qiskit_convert.py]: done in [use numpy.pi](https://github.com/CQCL/pytket-qiskit/pull/245/commits/143a76e6223d91df42a61a5267e24c631042e9ed)
[143a76e](https://github.com/CQCL/pytket-qiskit/pull/245/commits/143a76e6223d91df42a61a5267e24c631042e9ed)

File: pytket/extensions/qiskit/qiskit_convert.py [hunk 5]
```diff
@@ -653,22 +666,31 @@ def append_tk_command_to_qiskit(
...
-                raise NotImplementedError(
-                    "OpenQASM conditions must be an entire register in order"
-                )
         instruction = append_tk_command_to_qiskit(
             op.op, args[width:], qcirc, qregmap, cregmap, symb_map, range_preds  # type: ignore
         )
+        if len(cregmap[regname]) == width:
+            for i, a in enumerate(args[:width]):
+                if a.index != [i]:
+                    raise NotImplementedError(
```
[the remark this answers is not in the document]
Comment (cqc-melf) [reply pytket/extensions/qiskit/qiskit_convert.py:678]: Yes you are right, the conversion is not using qasm, so I have removed the qasm part of the error messages.

Done in @cqc-melf
[remove qasm from error message](https://github.com/CQCL/pytket-qiskit/pull/245/commits/720f522bc2593a07b5fedf89ed846a92c9c43bc7)
[720f522](https://github.com/CQCL/pytket-qiskit/pull/245/commits/720f522bc2593a07b5fedf89ed846a92c9c43bc7)

**Pull request merged**